src/components/networks/networkOverviewTab.jsx

```diff
@@ -149,3 +201,93 @@ export const NetworkOverviewTab = ({ network }) => {
+    const [ipAddress, setIpAddress] = useState('');
+    const [macAddress, setMacAddress] = useState('');
+    const [error, setError] = useState();
+
+    const add = () => {
+        setIsSubmitted(true);
+
+        if (!ipAddress || !macAddress)
+            return;
+
+        return networkAddStaticHostEntries({
+            connectionName: network.connectionName,
+            networkName: network.name,
+            macAddress,
+            ipAddress,
+            parentIndex,
+        })
+                .then(
+                    () => {
+                        setIsOpen(false);
+                        setIpAddress('');
+                        setMacAddress('');
+                        setError(undefined);
+                        networkGet({ connectionName: network.connectionName, id: network.id, updateOnly: true });
+                    },
+                    exc => setError(exc.message)
+                );
+    };
+
+    let addEntry = (
+        <Button variant="link"
+                id={idPrefix + "-static-host-entries-add"}
+                isDisabled={network.active}
+                isInline
+                onClick={() => setIsOpen(true)}>
+            {_("add entry")}
+        </Button>
+    );
+    if (network.active) {
+        addEntry = (
+            <Tooltip content={_("Deactivate the network in order to edit it")}>
+                {addEntry}
+            </Tooltip>
+        );
+    }
+
+    return (
+        <>
+            {addEntry}
+            <Modal position="top" variant="small" isOpen={isOpen} onClose={() => setIsOpen(false)}
+                title={_("Add a DHCP static host entry")}
+                footer={
+                    <>
+                        {error && <ModalError dialogError={_("Failed to save network settings")} dialogErrorDetail={error} />}
+                        <Button variant='secondary' onClick={add} id="add-new-static-entry-save">
+                            {_("Add")}
+                        </Button>
+                        <Button variant='link' id="add-new-static-entry-cancel" onClick={() => setIsOpen(false)}>
+                            {_("Cancel")}
+                        </Button>
+                    </>
+                }>
+                <Form onSubmit={e => {
+                    e.preventDefault();
+                }} isHorizontal>
+                    <FormGroup label={_("MAC address")} fieldId="add-new-static-entry-mac-address"
+                               helperTextInvalid={_("MAC address must not be empty")}
+                               validated={isSubmitted && !macAddress ? "error" : "default"}>
+                        <TextInput id="add-new-static-entry-mac-address"
+                                   validated={isSubmitted && !macAddress ? "error" : "default"}
+                                   value={macAddress}
+                                   onChange={setMacAddress} />
+                    </FormGroup>
```

Comment:
Also was there actually any specific reason why you decided to use MAC address even for IPv4? Since you can also identify the host by simple hostname (e.g. www.example.org), I think it's more convenient to let user configure entry identified by hostname instead of MAC (Or maybe have both MAC input and hostname input and let user choose how they want to identify the host).
But in short, if we want to have only one way of host identification, then I think hostname is more preferable to me than mac address.

Reply:
You can not have host elements without MAC for ipv4. Name attribute is for the DHCP to assign a specific hostname to the VM with the given mac address. So that it does not get the usual transient hostname.

Reply:
Right, DHCP+DNS work exactly the other way around :grin: MAC → IP → NAME. Of course it would be much more convenient if the UI already knew the MAC address -- this configuration is specific to a particular (virtual) interface, is it not? So how much freedom does the user actually have here? Do they have to know the one-and-only correct value, or does this somehow simultaneously define the MAC and the IP for that interface?